samples: Bluetooth: Broadcast Audio Source: Fix potential memory leak

FARLY7 opened the pull request:
In error cases, where send_data() returns early, the allocated net_buf is not free'd, causing a memory leak affecting the net_buf pool. Fix this by freeing the allocated net_buf before returning early from send_data().

Thalley reviewed (dismissed):
Thanks. This looks like it fixes an issue indeed. I think we can reduce the number of calls to `net_buf_unref(buf);` by moving `buf = net_buf_alloc(&tx_pool, K_FOREVER);` to just before `net_buf_add_mem`

FARLY7 commented:
> Thanks. This looks like it fixes an issue indeed. I think we can reduce the number of calls to `net_buf_unref(buf);` by moving `buf = net_buf_alloc(&tx_pool, K_FOREVER);` to just before `net_buf_add_mem`

Yeah, I agree. I'll redo the commit with this change instead.

FARLY7 committed:
samples: Bluetooth: Broadcast Audio Source: Fix potential memory leak

In error cases, where send_data() returns early, the allocated net_buf
is not free'd, causing a memory leak affecting the net_buf pool.
Fix this by freeing the allocated net_buf before returning early
from send_data().

Signed-off-by: Sean Farrelly <sean.farrelly@outlook.com>

FARLY7 commented:
> > Thanks. This looks like it fixes an issue indeed. I think we can reduce the number of calls to `net_buf_unref(buf);` by moving `buf = net_buf_alloc(&tx_pool, K_FOREVER);` to just before `net_buf_add_mem`
> 
> Yeah, I agree. I'll redo the commit with this change instead.

Because of the nearby `#ifdef` statements, moving the `net_buf alloc` to before the two `net_buf_add_mem` paths results in duplicating the `alloc` and `reserve` code. I opted just to revert back to the original fix of adding `net_buf_free` before the early return statements.

kartben commented:
@FARLY7 you need to sign off your commit, please.

```
git commit --amend --signoff
```

You may then force push to your branch and this should make our compliance check script happy :) Thanks!

FARLY7 commented:
> @FARLY7 you need to sign off your commit, please.
> 
> ```
> git commit --amend --signoff
> ```
> 
> You may then force push to your branch and this should make our compliance check script happy :) Thanks!

I'm not sure why it was missing on the second push. Fixed now :)
